packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -656,15 +656,38 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     final bool isShortcutModifierPressed = isMacOS ? keyEvent.isMetaPressed : keyEvent.isControlPressed;
     if (_movementKeys.contains(key)) {
       _handleMovement(key, wordModifier: isWordModifierPressed, lineModifier: isLineModifierPressed, shift: keyEvent.isShiftPressed);
+    } else if (key == LogicalKeyboardKey.delete) {
+      _handleDelete(forward: true);
+    } else if (key == LogicalKeyboardKey.backspace) {
+      _handleDelete(
+        forward: false, 
+        isWordModifierPressed: isWordModifierPressed, 
+        isLineModifierPressed: isLineModifierPressed
+      );
     } else if (isShortcutModifierPressed && _shortcutKeys.contains(key)) {
       // _handleShortcuts depends on being started in the same stack invocation
       // as the _handleKeyEvent method
       _handleShortcuts(key);
-    } else if (key == LogicalKeyboardKey.delete) {
-      _handleDelete(forward: true);
-    } else if (key == LogicalKeyboardKey.backspace) {
-      _handleDelete(forward: false);
+    } 
+  }
+
+  // Return the offset at the start of the nearest word to the left of the given
+  // offset.
+  static int _getLeftByWord(TextPainter textPainter, int offset, [bool includeWhitespace = true]) {
```

Comment:
I noticed that [this PR](https://github.com/flutter/flutter/pull/75032/files#diff-49a445da2fd4112ade71a9423b3acffca8d7ffcf694849159f0adb0ed40ba0d5R779) already dealt with block navigation with Editable fields, so I copied this method from there.